orttraining/orttraining/python/training/_module_output_transformation.py

```diff
@@ -0,0 +1,141 @@
+        return key_value_pairs
+
+    key_value_pairs = _key_value_pairs_from_output(user_output_names, user_outputs)
+    if issubclass(user_output_type, abc.Mapping):
+        return user_output_type(key_value_pairs)
+    elif issubclass(user_output_type, abc.Sequence):
+        try:
+            # Try constructing the user named tuple from the output tuple
+            return user_output_type(*[user_output for _, user_output in key_value_pairs])
+        except TypeError:
+            # The expected output type is not a namedtuple, but is a regular tuple type
+            pass
+
+    return key_value_pairs[0][1] if len(key_value_pairs) == 1 \
+        else tuple(user_output for _, user_output in key_value_pairs)
+
+def _transform_to_flat_structure(output):
+    def _transform_output_from_mapping(output_dict):
+        transformed_output = OrderedDict()
+        for key, value in output_dict.items():
+            if isinstance(value, torch.Tensor):
+                processed_key = _process_output_name(key)
+                transformed_output[processed_key] = value
+            elif isinstance(value, abc.Sequence):
+                for i, sequence_value in enumerate(value):
+                    if not isinstance(sequence_value, torch.Tensor):
+                        raise TypeError('ORTModule does not support the following output type {} \
+                            within a Sequence within a Mapping'.format(type(sequence_value)))
+                    processed_key = _process_output_name(key, i)
+                    transformed_output[processed_key] = sequence_value
+            else:
+                raise TypeError('ORTModule does not support the following output type {} within a Mapping'.format(type(value)))
+        return transformed_output
+
+    def _transform_output_from_sequence(output_sequence):
+        transformed_output = []
+        for value in output_sequence:
+            if isinstance(value, torch.Tensor):
+                transformed_output.append(value)
+            elif isinstance(value, abc.Sequence):
+                for sequence_value in value:
+                    if not isinstance(sequence_value, torch.Tensor):
+                        raise TypeError('ORTModule does not support the following output type {} \
+                            within a Sequence within a Sequence'.format(type(sequence_value)))
+                    transformed_output.append(sequence_value)
+            else:
+                raise TypeError('ORTModule does not support the following output type {} within a Sequence'.format(type(value)))
+        return tuple(transformed_output)
+
+    output_structure = None
+    if isinstance(output, abc.Mapping):
+        output_structure = _transform_output_from_mapping(output)
+    elif isinstance(output, abc.Sequence):
+        output_structure = _transform_output_from_sequence(output)
+    return output_structure
+
+class DerivedModule(torch.nn.Module):
```

Comment:
```suggestion
class _OriginalModuleFlattened(torch.nn.Module):
```

`DerivedModule` doesnt expose the intent of the class (what it does) and is presented as a public class because it doesnt start with `_`